streams/runners/storm/runtime/src/main/java/com/hortonworks/streamline/streams/runtime/storm/spout/AvroStreamsSnapshotDeserializer.java

```diff
@@ -65,7 +65,10 @@ private Object convertValue(Object deserializedObj) {
             List<Schema.Field> fields = indexedRecord.getSchema().getFields();
             ImmutableMap.Builder<String, Object> keyValues = ImmutableMap.builder();
             for (Schema.Field field : fields) {
-                keyValues.put(field.name(), convertValue(indexedRecord.get(field.pos())));
+                Object currentValue = convertValue(indexedRecord.get(field.pos()));
+                if (currentValue != null) {
```

Comment:
@priyank5485 User may have logic in UDFs/UDAFs or elsewhere having a check like map contains key which returns false though it should returned true, it is going to break that logic. We should not be filtering any of the received payload unless there is a strong reason todo. Here, I do not see any reason for filtering and send the received payload as it is by source to subsequent processors.

 with this change we are converting the received payload 
```
{
"id": 12322,
"name": "foo",
location: null
}
```
into 
```
{
"id": 12322,
"name":"foo",
}
```
which are not equal.

When avro schema is defined as 
```
{
 "namespace":"hwx",
 "type":"record",
 "name":"",
 "fields":[
    ...
    { "name": "location", "type": [ "string", "null" ] },
    ...
 ]
}
```
then `value` field must exist either as some string value or as null but it should not be missing. With this fix, it would break that constraint and result in error.

>We need this map to be immutable since its going to be sent across in storm. I am still not clear on exact solution you are suggesting? Is there an immutable map that allows null values?

I have mentioned in my earlier [comment](https://github.com/hortonworks/streamline/pull/775#discussion_r125385744) about creating HashMap and return unmodifiable map for immutability which can be done like below.
```
        Map<String, Object> fieldValues = new HashMap<>();
        Object values = convertValue(deserializedObj);
        if (values instanceof Map) {
            fieldValues.putAll((Map) values);
        } else {
            fieldValues.put(StreamlineEvent.PRIMITIVE_PAYLOAD_FIELD, values);
        }

        return Collections.unmodifiableMap(fieldValues);
```

Issue encountered by user:
there was a null value sent by client as part of some field which is defined in schema as it can be either null or non-null value and the current code  is trying to set that but it is throwing NPE because of using `ImmutableMap`. 
So, we should fix that instead of filtering out those valid null values, one way to fix that is as above.